src/app/core/entity/entity-actions/entity-anonymize.service.spec.ts

```diff
@@ -0,0 +1,299 @@
+import { TestBed } from "@angular/core/testing";
+import { EntityMapperService } from "../entity-mapper/entity-mapper.service";
+import { Entity } from "../model/entity";
+import { of } from "rxjs";
+import { DatabaseEntity } from "../database-entity.decorator";
+import { DatabaseField } from "../database-field.decorator";
+import {
+  mockEntityMapper,
+  MockEntityMapperService,
+} from "../entity-mapper/mock-entity-mapper-service";
+import { comparableEntityData } from "../../../utils/expect-entity-data.spec";
+import { UpdateMetadata } from "../model/update-metadata";
+import { FileService } from "../../../features/file/file.service";
+import { CoreTestingModule } from "../../../utils/core-testing.module";
+import { DefaultDatatype } from "../default-datatype/default.datatype";
+import { FileDatatype } from "../../../features/file/file.datatype";
+import moment from "moment";
+import {
+  allEntities,
+  ENTITIES,
+  EntityWithRelations,
+  expectAllUnchangedExcept,
+} from "./entity-actions-test-utils";
+import { EntityAnonymizeService } from "./entity-anonymize.service";
+
+describe("EntityAnonymizeService", () => {
+  let service: EntityAnonymizeService;
+  let entityMapper: MockEntityMapperService;
+  let mockFileService: jasmine.SpyObj<FileService>;
+
+  beforeEach(() => {
+    entityMapper = mockEntityMapper(allEntities.map((e) => e.copy()));
+
+    mockFileService = jasmine.createSpyObj(["removeFile"]);
+    mockFileService.removeFile.and.returnValue(of(null));
+
+    TestBed.configureTestingModule({
+      imports: [CoreTestingModule],
+      providers: [
+        EntityAnonymizeService,
+        { provide: EntityMapperService, useValue: entityMapper },
+        { provide: FileService, useValue: mockFileService },
+        { provide: DefaultDatatype, useClass: FileDatatype, multi: true },
+      ],
+    });
+
+    service = TestBed.inject(EntityAnonymizeService);
+  });
+
+  /*
+   * ANONYMIZATION
+   */
+  @DatabaseEntity("AnonymizableEntity")
+  class AnonymizableEntity extends Entity {
+    static override hasPII = true;
+
+    @DatabaseField() defaultField: string;
+
+    @DatabaseField({ anonymize: "retain" })
+    retainedField: string;
+
+    @DatabaseField({
+      anonymize: "retain-anonymized",
+      dataType: "array",
+      innerDataType: "date-only",
+    })
+    retainAnonymizedDates: Date[];
+
+    @DatabaseField({ dataType: "file" }) file: string;
+
+    @DatabaseField({ anonymize: "retain-anonymized", dataType: "entity-array" })
+    referencesToRetainAnonymized: string[];
+
+    static create(properties: Partial<AnonymizableEntity>) {
+      return Object.assign(new AnonymizableEntity(), properties);
+    }
+
+    static expectAnonymized(
+      entityId: string,
```

Comment:
Is there a reason why you don't just pass the entity over?

Reply:
I want an implicit check that the actual entity is not just a return value but really stored in the database